python-runnables/resize-node-pool/runnable.py

```diff
@@ -31,22 +31,30 @@ def run(self, progress_callback):
         
         node_pool_id = self.config.get('nodePoolId', None)
         node_pool = None
-        for profile in cluster.agent_pool_profiles:
-            if profile.name == node_pool_id or (_is_none_or_blank(node_pool_id) and len(cluster.agent_pool_profiles) == 1):
+        node_pools = [node_pool for node_pool in clusters.agent_pools.list(resource_group, cluster_name)]
+        if _is_none_or_blank(node_pool_id) and len(node_pools) > 1:
+            raise Exception("Cluster has %s node pools, you need to specify the node pool id'" % len(node_pools))
+        for profile in node_pools:
+            if profile.name == node_pool_id or (_is_none_or_blank(node_pool_id) and len(node_pools) == 1):
                 node_pool = profile
         if node_pool is None:
             raise Exception("Unable to find node pool '%s'" % (node_pool_id))
 
         desired_count = self.config['numNodes']
         logging.info("Resize to %s" % desired_count)
         if desired_count == 0:
-            raise Exception("Can't delete node pool '%s'" % (node_pool_id))
+            def do_delete():
+                cluster_update_op = clusters.agent_pools.begin_delete(resource_group, cluster_name, node_pool_id)
```

Comment:
`node_pool_id` will still be `None` here if it wasn't set. You probably need to do a `if node_pool_id is None: node_pool_id = node_pool.name` after selecting the `node_pool`.